scalableminds/webknossos
# Marionette upgrade to v3

Description of changes:
- This PR upgrades marionette to v3. I've resolved almost all deprecation warnings logged by the marionette-v3-compat tool by migrating the code.
- The only missing one is `Deprecation warning: CompositeView is deprecated. Convert to View at your earliest convenience`. Currently I can find 18 files that are using a CompositeView, so this would be a significant amount of work imo. However the CompositeView is "only" deprecated, so it would still work in marionette v3, even without the compatibility tool. So maybe we should leave them as they are for now.

Most notable changes in v3:
- ItemView/LayoutView: Marionette.ItemView/Marionette.LayoutView is now Marionette.View in v3
- RegionShowEvent: Showing a view in a region no longer triggers a show event on the View in v3
- templateHelpers: View's templateHelpers is now templateContext in v3
- Regions are not attached directly to the layout. Use getRegion to access the region or showChildView to show a View within it.
- app.addInitializer was removed

Steps to test:
- Test all different functionality that could be affected by a marionette upgrade and check whether any marionette deprecation warnings (apart from the CompositeView one) are logged in the console

Issues:
- needed by #1373 as only marionette v3 works with jquery v3

---
- [x] Removed marionette-v3-compat module before deploying to production


<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/1486/create?referer=github" target="_blank">Log Time</a>
